pkg/controller/cassandradatacenter/statefulset.go

```diff
@@ -165,7 +212,7 @@ func newSysctlLimitsContainer(cdc *cassandraoperatorv1alpha1.CassandraDataCenter
 		Image:           cdc.Spec.CassandraImage,
 		ImagePullPolicy: cdc.Spec.ImagePullPolicy,
 		SecurityContext: &corev1.SecurityContext{
-			Privileged: func() *bool { b := true; return &b }(),
+			Privileged: boolRef(cdc.Spec.PrivilegedSupported),
```

Comment:
Is there a point in running the sysctl init container unprivileged? Shouldn't we completely drop the container if `cdc.Spec.PrivilegedSupported == false`?

Reply:
That's a very good question. In Java version we did [this](https://github.com/instaclustr/cassandra-operator/blob/master/java/operator/src/main/java/com/instaclustr/cassandra/operator/controller/DataCenterReconciliationController.java#L340), so I assumed we might want to do this here too. The cluster seems to be working fine in my local testing, but I'm not sure it actually tests this flow properly.

Reply:
Here is what happens when the sysctl container runs unprivileged:

```
kc logs -f cassandra-test-dc-cassandra-0 -c sysctl-limits 
+ sysctl -w vm.max_map_count=1048575
sysctl: permission denied on key 'vm.max_map_count'
+ true
```

The pod starts fine because we're doing `|| true`, however it has no effect in this case as it cannot modify the setting on the host.

AFAICT we can drop this container when we can't run as privileged.